src/middlewares/cart.middleware.js

```diff
@@ -0,0 +1,74 @@
+import { CartService } from '../services/cart.service';
+import { ProductService } from '../services/product.service';
+
+export const getUserCart = async (req, res, next) => {
+  try {
+    const { id: userId } = req.user;
+    const { productId } = req.body;
+
+    let userCart = await CartService.getCart(userId);
+    if (!userCart) {
+      const total = 0;
+      const products = [{ quantity: 1, productId: productId }];
+      const cart = {
+        userId,
+        total,
+        products,
+      };
+      userCart = await CartService.createCart(cart);
+    } else {
+      let products = userCart.products;
+      let existingProduct = products.find(p => p.productId === productId);
+      if (existingProduct) {
+        existingProduct.quantity += 1;
+      } else {
+        products.push({ quantity: 1, productId: productId });
```

Comment:
you need to check the condition of the added product when quantity>1

Reply:
Here,
You mean, like when user adds quantity?
Currently what am doing is, if a user adds a product that's already in the cart, I increment quantity by one